test/e2e/user_quota_tracing/user_quota_tracing_test.go

```diff
@@ -0,0 +1,261 @@
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package user_quota_tracing_test
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/apache/yunikorn-core/pkg/common/configs"
+	tests "github.com/apache/yunikorn-k8shim/test/e2e"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/common"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/k8s"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/yunikorn"
+	siCommon "github.com/apache/yunikorn-scheduler-interface/lib/go/common"
+)
+
+const (
+	NANESPACE_LENGTH  = 10
+	WAIT_INTERVAL     = 60
+	DEFAULT_PARTITION = "default"
+)
+
+var _ = Describe("QuotaTracking: Two leaf queus for two groups", func() {
+	var ns string
+	BeforeEach(func() {
+		ns = "ns-" + common.RandSeq(NANESPACE_LENGTH)
+		By(fmt.Sprintf("Create namespace: %s", ns))
+		var ns1, err1 = kClient.CreateNamespace(ns, nil)
+		Ω(err1).NotTo(HaveOccurred())
+		Ω(ns1.Status.Phase).To(Equal(v1.NamespaceActive))
+	})
+
+	It("User qauta trace with 3 users and 2 groups", func() {
+		groups := []string{"group1", "group2"}
+		users := []string{"user1", "user2", "user3"}
+		queuePath := []string{"root", "root.group1_resources", "root.group2_resources"}
+		zeroPod := map[string]int64{siCommon.CPU: 0, siCommon.Memory: 0}
+		onePod := map[string]int64{siCommon.CPU: 100, siCommon.Memory: 50}
+		twoPods := map[string]int64{siCommon.CPU: 200, siCommon.Memory: 100}
+		yunikorn.UpdateCustomConfigMapWrapper(oldConfigMap, "", annotation, func(sc *configs.SchedulerConfig) error {
+			// remove placement rules so we can control queue
+			sc.Partitions[0].PlacementRules = nil
+			queuesConfigs := []struct {
+				partition, parentQueue, QueueName string
+			}{
+				{DEFAULT_PARTITION, queuePath[0], "group1_resources"},
+				{DEFAULT_PARTITION, queuePath[0], "group2_resources"},
+			}
+			for _, queueConfig := range queuesConfigs {
+				By(fmt.Sprintf("Add child queue %s to the parent queue %s", queueConfig.QueueName, queueConfig.parentQueue))
+				config := configs.QueueConfig{
+					Name: queueConfig.QueueName,
+				}
+				if err := common.AddQueue(sc, queueConfig.partition, queueConfig.parentQueue, config); err != nil {
+					return err
+				}
+			}
+			return nil
+		})
+
+		/*
+		*  groups: group1, group2
+		*  users: user1, user2, user3
+		*  user1 -> group1
+		*  user2 -> group2
+		*  user3 -> group1, group2
+		 */
+		configs := []struct {
+			AppID, Queue, UserName string
+			GroupsNames            []string
+		}{
+			{"teacher-app-01", queuePath[1], users[0], []string{groups[0]}},
+			{"students-app-01", queuePath[2], users[1], []string{groups[1]}},
+			{"assistant-app-01", queuePath[2], users[2], []string{groups[1]}},
+			{"assistant-app-02", queuePath[1], users[2], []string{groups[0]}},
```

Comment:
If these are strings which do not change inside the slice, I'd prefer referencing them as literals rather than slice elements. This is difficult to interpret. Extract them to const if that helps.